src/tokens/ERC5095.sol

```diff
@@ -239,7 +341,7 @@ contract ERC5095 is ERC20Permit, IERC5095 {
             underlying,
             maturity,
             assets,
-            Cast.u128(s - (s / 100))
+            0
```

Comment:
Leave the rest of the method as is, but just to make it parallel with the other mint method:

```
uint128 returned = IMarketPlace(marketplace).buyPrincipalToken(
            underlying,
            maturity,
            s,
            assets)
```

Reply:
This buys principal tokens for `s`, and doesnt account for slippage given youre using `assets` which you just calculated for the "slippage" control.